nym-api/src/node_describe_cache/mod.rs

```diff
@@ -79,17 +78,16 @@ impl NodeDescriptionProvider {
 }
 
 async fn try_get_client(
-    gateway: &Gateway,
+    host: &str,
+    identity_key: &IdentityKey,
 ) -> Result<nym_node_requests::api::Client, NodeDescribeCacheError> {
-    let gateway_host = &gateway.host;
-
     // first try the standard port in case the operator didn't put the node behind the proxy,
     // then default https (443)
     // finally default http (80)
     let addresses_to_try = vec![
-        format!("http://{gateway_host}:{DEFAULT_NYM_NODE_HTTP_PORT}"),
-        format!("https://{gateway_host}"),
-        format!("http://{gateway_host}"),
+        format!("http://{host}:{DEFAULT_NYM_NODE_HTTP_PORT}"),
```

Comment:
if you're using the same list of addresses for both gateways and mixnodes, you will have to add port `8000` (`DEFAULT_HTTP_API_LISTENING_PORT`) because old, migrated, mixnodes will most likely be using that one instead; or if you want to be fancy, you could add an extra argument `port: Option<u16>` since mixnodes will have announced this one to the directory server

Reply:
- brand new nym-node will be on `8080` (or `80`/`443` if the user bothered to setup reverse proxy)
- migrated gateway will also be on `8080`
- migrated mixnode, however, will use whatever they have configured before. by default it was `8000` but the "true" value is stored in the mixnet contract

Reply:
@jstuczyn could we count on what the mixnodes announce completly, ie remove the DEFAULT_PORT, and only leave what is announced and `80/443`